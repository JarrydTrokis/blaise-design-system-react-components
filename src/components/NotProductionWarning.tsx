import React, {ReactElement} from "react";


function NotProductionWarning(): ReactElement {
    return (
        <>
            <div className="census-warning" style={{background: "#222", color: "#222"}}>
                <div className="container">
                    <div className="panel panel--warn-branded panel--no-title" style={{background: "#222"}}>
                        <span className="panel__icon" aria-hidden="true" style={{color: "#222"}}>!</span>
                        <div className="panel__body">
                            <p className="u-mb-no">This environment is not a production environment. Do not upload any live data to this service.</p>
                        </div>
                    </div>
                </div>
            </div>
        </>
    );
}

export default NotProductionWarning;
