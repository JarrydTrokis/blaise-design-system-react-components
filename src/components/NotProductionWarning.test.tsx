import { render, screen } from "@testing-library/react";
import React from "react";
import NotProductionWarning from "./NotProductionWarning";

describe("ONS Not production warning Test", () => {
    it("matches Snapshot", () => {
        expect(render(<NotProductionWarning />)).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const { container } = render(<NotProductionWarning />);
        expect(container).toBeDefined();
    });

    it("should display warning paragraph text", () => {
        render(<NotProductionWarning />);
        expect(screen.getByText(/This environment is not a production environment. Do not upload any live data to this service./)).toBeVisible();
    });
});
