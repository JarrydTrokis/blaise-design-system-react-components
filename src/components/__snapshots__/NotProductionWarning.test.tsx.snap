// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Not production warning Test matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="background: rgb(34, 34, 34); color: rgb(34, 34, 34);"
      >
        <div
          class="ons-container"
        >
          <div
            class="ons-panel ons-panel--warn-branded ons-panel--no-title"
            style="background: rgb(34, 34, 34);"
          >
            <span
              aria-hidden="true"
              class="ons-panel__icon"
              style="color: rgb(34, 34, 34);"
            >
              !
            </span>
            <span
              class="ons-panel__assistive-text ons-u-vh"
            >
              Warning: 
            </span>
            <div
              class="ons-panel__body"
            >
              <p
                class="ons-u-mb-no"
              >
                This environment is not a production environment. Do not upload any live data to this service.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="background: rgb(34, 34, 34); color: rgb(34, 34, 34);"
    >
      <div
        class="ons-container"
      >
        <div
          class="ons-panel ons-panel--warn-branded ons-panel--no-title"
          style="background: rgb(34, 34, 34);"
        >
          <span
            aria-hidden="true"
            class="ons-panel__icon"
            style="color: rgb(34, 34, 34);"
          >
            !
          </span>
          <span
            class="ons-panel__assistive-text ons-u-vh"
          >
            Warning: 
          </span>
          <div
            class="ons-panel__body"
          >
            <p
              class="ons-u-mb-no"
            >
              This environment is not a production environment. Do not upload any live data to this service.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
